controllers/manageddatabase_controller.go

```diff
@@ -190,10 +247,11 @@ func (c *ManagedDatabaseController) ReconcileManagedDatabase(req ctrl.Request) (
 }
 
 type migrationContext struct {
-	ctx     context.Context
-	log     logr.Logger
-	db      *dba.ManagedDatabase
-	version *dba.DatabaseMigration
+	ctx           context.Context
```

Comment:
Just because it has the word context? I think this was probably actually a bad idea. I hated how long the signatures were getting on the methods to refactored some state out into an "args bundle". It doesn't get re-used enough to be worthwhile IMO. Refactoring ideas?